code/modules/station_goals/bsa.dm

```diff
@@ -0,0 +1,408 @@
+	reload_cooldown = 100
+	
+/obj/machinery/bsa/full/admin/east
+	icon_state = "cannon_east"
+	cannon_direction = EAST
+
+/obj/machinery/bsa/full/proc/get_front_turf()
+	switch(dir)
+		if(WEST)
+			return locate(x - 6,y,z)
+		if(EAST)
+			return locate(x + 4,y,z)
+	return get_turf(src)
+
+/obj/machinery/bsa/full/proc/get_back_turf()
+	switch(dir)
+		if(WEST)
+			return locate(x + 4,y,z)
+		if(EAST)
+			return locate(x - 6,y,z)
+	return get_turf(src)
+
+/obj/machinery/bsa/full/proc/get_target_turf()
+	switch(dir)
+		if(WEST)
+			return locate(1,y,z)
+		if(EAST)
+			return locate(world.maxx,y,z)
+	return get_turf(src)
+
+/obj/machinery/bsa/full/New(loc, direction)
+	..()
+
+	if(direction)
+		cannon_direction = direction
+
+	switch(cannon_direction)
+		if(WEST)
+			dir = WEST
+			pixel_x = -192
+			top_layer = image("icons/obj/lavaland/orbital_cannon.dmi", "top_west")
+			top_layer.layer = 4.1
+			icon_state = "cannon_west"
+		if(EAST)
+			dir = EAST
+			top_layer = image("icons/obj/lavaland/orbital_cannon.dmi", "top_east")
+			top_layer.layer = 4.1
+			icon_state = "cannon_east"
+	overlays += top_layer
+	reload()
+
+/obj/machinery/bsa/full/proc/fire(mob/user, turf/bullseye)
+	var/turf/point = get_front_turf()
+	for(var/turf/T in getline(get_step(point,dir),get_target_turf()))
+		T.ex_act(1)
+		for(var/atom/A in T)
+			A.ex_act(1)
+
+	point.Beam(get_target_turf(), icon_state = "bsa_beam", time = 50, maxdistance = world.maxx, beam_type = /obj/effect/ebeam/deadly) //ZZZAP
+	playsound(src, 'sound/machines/bsa_fire.ogg', 100, 1)
+	
+	message_admins("[key_name_admin(user)] has launched an artillery strike.")
+	explosion(bullseye,ex_power,ex_power*2,ex_power*4)
+
+	reload()
+
+/obj/machinery/bsa/full/proc/reload()
+	use_power(power_used_per_shot)
+	last_fire_time = world.time / 10
+
+/obj/item/weapon/circuitboard/machine/bsa/back
+	name = "Bluespace Artillery Generator (Machine Board)"
+	build_path = /obj/machinery/bsa/back
+	origin_tech = "engineering=2;combat=2;bluespace=2" //No freebies!
+	req_components = list(
+							/obj/item/weapon/stock_parts/capacitor/quadratic = 5,
+							/obj/item/stack/cable_coil = 2)
+
+/obj/item/weapon/circuitboard/machine/bsa/middle
+	name = "Bluespace Artillery Fusor (Machine Board)"
+	build_path = /obj/machinery/bsa/middle
+	origin_tech = "engineering=2;combat=2;bluespace=2"
+	req_components = list(
+							/obj/item/weapon/ore/bluespace_crystal = 20,
+							/obj/item/stack/cable_coil = 2)
+
+/obj/item/weapon/circuitboard/machine/bsa/front
+	name = "Bluespace Artillery Bore (Machine Board)"
+	build_path = /obj/machinery/bsa/front
+	origin_tech = "engineering=2;combat=2;bluespace=2"
+	req_components = list(
+							/obj/item/weapon/stock_parts/manipulator/femto = 5,
+							/obj/item/stack/cable_coil = 2)
+
+/obj/item/weapon/circuitboard/computer/bsa_control
+	name = "Bluespace Artillery Controls (Computer Board)"
+	build_path = /obj/machinery/computer/bsa_control
+	origin_tech = "engineering=2;combat=2;bluespace=2"
```

Comment:
Huh, this is something just any schmoo can build with requisite materials?

I like it 😈